packages/cli/src/cog/cog.stac.job.ts

```diff
@@ -0,0 +1,416 @@
+    CogStacKeywords,
+    CogStacLicense,
+    CogStacVersion,
+    StacCollection,
+    StacLink,
+    StacProvider,
+    CogStac,
+    CogStacItemExtensions,
+    CogStacItem,
+} from './stac';
+import {
+    CogBuilderMetadata,
+    CogJob,
+    CogJobJson,
+    CogOutputProperties,
+    CogSourceProperties,
+    FeatureCollectionWithCrs,
+} from './types';
+
+export const MaxConcurrencyDefault = 50;
+
+export interface JobCreationContext {
+    /** Source config */
+    sourceLocation: FileConfig | FileConfigPath;
+
+    /** Output config */
+    outputLocation: FileConfig;
+
+    /** Should the imagery be cut to a cutline */
+    cutline?: {
+        href: string;
+        blend: number;
+    };
+
+    targetProjection: ProjectionTileMatrixSet;
+
+    override?: {
+        /** Override job id */
+        id?: string;
+
+        /**
+         * Image quality
+         * @default GdalCogBuilderDefaults.quality
+         */
+        quality?: number;
+
+        /**
+         * Number of threads to use for fetches
+         * @default MaxConcurrencyDefault
+         */
+        concurrency?: number;
+
+        /**
+         * Override the source projection
+         */
+        projection?: Epsg;
+
+        /**
+         * Resampling method
+         * @Default  GdalCogBuilderDefaults.resampling
+         */
+        resampling?: GdalCogBuilderResampling;
+    };
+
+    /**
+     * Should this job be submitted to batch now?
+     * @default false
+     */
+    batch: boolean;
+
+    /** Should this job ignore source coverage and just produce one big COG for EPSG extent */
+    oneCogCovering: boolean;
+}
+
+/**
+ * Attempt to parse a year from a imagery name
+ * @example wellington_urban_2017_0.10m -> 2017
+ * @param name Imagery name to parse
+ * @return imagery year, -1 for failure to parse
+ */
+export function extractYearRangeFromName(name: string): [number, number] {
+    const re = /(?:^|\D)(\d{4})(?:-(\d{2}))?(?:$|\D)/g;
+
+    const years: number[] = [];
+
+    for (let m = re.exec(name); m != null; m = re.exec(name)) {
+        years.push(parseInt(m[1]));
+        if (m[2] != null) years.push(parseInt(m[1].slice(0, 2) + m[2]));
+    }
+
+    if (years.length == 0) {
+        return [-1, -1];
+    }
+
+    years.sort();
+
+    return [years[0], years[years.length - 1] + 1];
+}
+
+/**
+ * Information needed to create cogs
+ */
+export class CogStacJob implements CogJob {
+    json: CogJobJson;
+    private cacheTargetZoom?: {
+        gsd: number;
+        zoom: number;
+    };
+
+    /**
+     * Load the job.json
+
+     * @param jobpath where to load the job.json from
+     */
+    static async load(jobpath: string): Promise<CogStacJob> {
+        return new CogStacJob(await FileOperator.readJson<CogJobJson>(jobpath));
+    }
+
+    /**
+     * Create job.json, collection.json, source.geojson, covering.geojson, cutlint.geojson.gz and
+     * stac descriptions of the target COGs
+
+     * @param id unqiue id for this imagery set
```

Comment:
These are not actually `@param`, since its a destructured object.

you only really have one parameter, `context` which is all of this stuff inside of it, to get this to work you would be best to extract this into its own interface and add comments to that.